### PR TITLE
add busyRetry for EditCommands that cannot finish immediately

### PR DESCRIPTION
**Pull request by kabentley**
This is a breaking change for the @alpha `EditCommand` class. I also promoted it to @beta.

@bbastings @swwilson-bsi, feedback appreciated.

### CODE REVIEW

**Review by bbastings — APPROVED** [review]

File: editor/backend/src/EditCommand.ts [hunk 1]
```diff
@@ -31,23 +31,39 @@ export class EditCommand implements EditCommandIpc {
...
 
-  public onFinish(): void { }
+  /**
+   * Called when another EditCommand wishes to become the active EditCommand.
+   * Subclasses should complete and save their work as soon as possible and then return "done".
+   * If it is not currently possible to finish, return any string other than "done" and the other EditCommand will have to wait and retry,
+   * potentially showing the returned string to the user.
+   */
+  public async requestFinish(): Promise<"done" | string> {
+    this.onFinish(); // TODO: temporary, remove
```
Comment (pmconne) [inline editor/backend/src/EditCommand.ts:54]: In this PR, or in a subsequent one?

Comment (kabentley) [reply editor/backend/src/EditCommand.ts:54]: i'm worried about substation and synchro

File: editor/backend/src/EditCommand.ts [hunk 2]
```diff
@@ -80,11 +98,19 @@ export class EditCommandAdmin {
...
+  public static async finishCommand() {
+    if (this._activeCommand) {
+      const finished = await this._activeCommand.requestFinish();
+      if ("done" !== finished)
+        throw new BackendError(IModelStatus.ServerTimeout, editorIpcStrings.commandBusy, finished);
+    }
+    this._activeCommand = undefined;
+  }
+
+  public static async runCommand(cmd: EditCommand): Promise<any> {
```
Comment (pmconne) [inline editor/backend/src/EditCommand.ts:110]: Are `runCommand` and `finishCommand` intended to be `@beta` now? If so please consider documenting them and other promoted APIs.

Comment (kabentley) [reply editor/backend/src/EditCommand.ts:114]: made them @internal, but added documentation to the ipc interfaces that are used to call them.

**Review by pmconne — APPROVED** [review]

**Pull request merged**